hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/resolveDirty.kt

```diff
@@ -49,7 +49,16 @@ private fun RuntimeInfo.resolveDirtyMethods(redefined: RuntimeInfo): List<Method
     return redefined.methodIndex.mapNotNull { (methodId, redefinedMethod) ->
         val previousMethod = methodIndex[methodId] ?: return@mapNotNull redefinedMethod
         if (previousMethod.rootScope.hash != redefinedMethod.rootScope.hash) {
-            return@mapNotNull redefinedMethod
+            if (redefinedMethod.rootScope.isComposeEntry) {
+                throw IllegalStateException(
+                    """
+                Compose Hot Reload does not support the redefinition of compose entry method $methodId.
```

Comment:
Some minor adjustments would be nice:

1) 'of the Compose entry method'
2) Please restart the App or revert the changes in '$methodId'